added ne_50m_physical_geography_marine_polys

jbusecke opened the pull request:
Adds ocean regions to the natural earth module.

jbusecke commented:
I also noticed that there are marine polys for 110m and 10m available. Should I add those aswell?

mathause commented on docs/defined_ocean_basins.rst:
```diff
@@ -0,0 +1,61 @@
+################################
+Marine Areas/Ocean Basins (NaturalEarth)
+################################
```
```suggestion
########################################
Marine Areas/Ocean Basins (NaturalEarth)
########################################
```

I think they need the same length as the title.

jbusecke commented:
I can confirm that they are different polygons! So they should not be removed. Either they should be renamed (e.g. West/East Ross Sea) or joined to a single polygon. Do you think it would be worth raising an issue upstream?

jbusecke commented on docs/defined_ocean_basins.rst:
```diff
@@ -0,0 +1,61 @@
+    LON_EDGE = np.arange(0, 360)
+    LAT_EDGE = np.arange(90, -90, -1)
+
+    plt.pcolormesh(LON_EDGE, LAT_EDGE, mask_ma, cmap='viridis');
```
Ah nice.

jbusecke commented:
I added the 110m, but noticed it has quite a lot of gaps, so I refrained from using it in the example. I think the 50m version is the much better choice. Also added the doc page to the index and updated tests.

mathause commented on regionmask/defined_regions/natural_earth.py:
```diff
@@ -213,5 +216,37 @@ def land_110(self):
+                abbrevs="name",
+            )
+
+            self._ocean_basins_50 = _obtain_ne(**opt)
```
For the Mediterranean I would do it like so:

```suggestion
            regs = _obtain_ne(**opt)
            
            # NOTE: naturalearth includes duplicate names

            # raise an error if naturalearth changes the name
            msg = "naturalearth renamed this region, please raise an issue in regionmask"
            assert regs[14].name == 'Mediterranean Sea' , msg

            regs[14].name = 'Mediterranean Sea East'                                                                                                                                                    
            regs[14].abbrev = 'Mediterranean Sea East'
            # dito for regs[30] 
            self._ocean_basins_50 = regs
```

jbusecke replied:
This seems like a great workaround until they fix it upstream.

jbusecke replied:
For the time beeing. Should we just do the same thing with the Ross Sea? Ill raise an issue upstream.

jbusecke replied:
Upstream issue here: https://github.com/nvkelso/natural-earth-vector/issues/319

mathause commented on docs/defined_ocean_basins.rst:
```diff
@@ -0,0 +1,58 @@
+They are automatically downloaded on-the-fly (but only once) with cartopy and opened with geopandas.
+The following marine regions are defined in regionmask:
+
+* Marine Areas 1:50m
```
```suggestion
* Marine Areas 1:50m
* Marine Areas 1:110m
```

jbusecke commented:
I have removed the 110 again, added renaming for both Ross and Med including the warning. 
Applying the merging dict yields this:
![image](https://user-images.githubusercontent.com/14314623/77207256-4a90ce80-6acf-11ea-9f05-2fd2d420e6da.png)
This is pretty much exactly what I needed! Thanks a lot for the help.

mathause commented on regionmask/defined_regions/natural_earth.py:
```diff
@@ -213,5 +215,44 @@ def land_110(self):
+            regs[26].abbrev = "Ross Sea Eastern Basin"
+            # dito for regs[30]
+            regs[29].name = "Ross Sea Western Basin"
+            regs[29].abbrev = "Ross Sea Western Basin"
```
```suggestion

           # rename the "Mediterranean Sea" region
            assert regs[14].name == "Mediterranean Sea", msg
            regs[14].name = "Mediterranean Sea Eastern Basin"
            regs[14].abbrev = "Mediterranean Sea Eastern Basin"

            assert regs[30].name == "Mediterranean Sea", msg
            regs[30].name = "Mediterranean Sea Western Basin"
            regs[30].abbrev = "Mediterranean Sea Western Basin"

           # rename the "Ross Sea" region
            assert regs[26].name == "Ross Sea", msg
            regs[26].name = "Ross Sea Eastern Basin"
            regs[26].abbrev = "Ross Sea Eastern Basin"
            
            assert regs[29].name == "Ross Sea", msg
            regs[29].name = "Ross Sea Western Basin"
            regs[29].abbrev = "Ross Sea Western Basin"
```

I'm a bit paranoid renaming something that we shouldn't so I'd like to have checks for all of them

jbusecke replied:
Oh yeah, even better.

jbusecke commented:
@mathause , do you think this is good to go? Or do you think we need some more changes? I also wanted to thank you for all the help implementing this!

mathause commented:
Looks good to me! Thanks a lot for your contribution @jbusecke

jbusecke commented:
Seems like the docs are not built yet. Is there a trigger for it?

jbusecke commented:
Ah it seems like it fails due to high memory consumption. Do you think this is due to changes I made?

mathause commented:
Thanks for the reminder & no, that's not your fault. I feel I spend more time getting readthedocs to run than working on the package... :( I'll have to ask for more ressources, I already reduced the usage of conda to the minimum I could - this did work 6 weeks ago...

jbusecke commented:
Ugh sorry to hear that...Let me know if I can do something.

mathause commented:
I got more resources and it works again - nice. I already fixed the white line and removed the colorbar, but I'll render that later.
